client/marklogicClient.ts

```diff
@@ -56,7 +62,7 @@ export class MlClientParameters {
     sameAs(other: MlClientParameters): boolean {
         return (
             this.host === other.host &&
-            this.port === other.port &&
+            ((isNaN(this.port) && isNaN(other.port)) || this.port === other.port) &&
```

Comment:
I usually prefer to do ports as strings instead of numbers for reasons like this.  Is there a reason to use numbers here?

Reply:
It's a property of the `MlClientParameters` class - I didn't want to try changing the class definition to make the test impact, given the larger blast radius of doing so. Just wanted to get the test passing, and that test for some reason has port=NaN, so this check is needed since NaN !== NaN.